enterprise/backup/src/main/java/org/neo4j/backup/BackupCopyService.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.backup;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Consumer;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import org.neo4j.commandline.admin.CommandFailed;
+import org.neo4j.commandline.admin.OutsideWorld;
+
+import static java.lang.String.format;
+
+public class BackupCopyService
+{
+    private static final int MAX_OLD_BACKUPS = 1000;
+
+    private final OutsideWorld outsideWorld;
+
+    public BackupCopyService( OutsideWorld outsideWorld )
+    {
+        this.outsideWorld = outsideWorld;
+    }
+
+    public void moveBackupLocation( File oldLocation, File newLocation ) throws CommandFailed
+    {
+        try
+        {
+            outsideWorld.fileSystem().renameFile( oldLocation, newLocation );
```

Comment:
This won't fly on block device. Shame we didn't notice this on the original implementation you copied this from.

Reply:
If there is a simple work around I can address it still, that shouldn't be an issue

Reply:
The `org.neo4j.com.storecopy.StoreCopyClient` has an example of how it could be done. But we are clearly also missing a block device test for the `neo4j-admin backup` command.

Reply:
The key is figuring out which files are stored on the block device, and then moving those files via the `FileMoveAction.copyViaPageCache`.

Reply:
Not addressed - will create a new ticket afterwards.